File: App/src/interfaces/Models/EduLinks/ViewModels/EduLinkViewModel.ts
import { UserRoles } from "enums/UserRoles";
import { CommentVm } from "interfaces/Models/Comments/CommentVm";
import { FileVm } from "interfaces/Models/FilesMetadata/ViewModels/FileVm";

export interface EduLinkViewModel {
  title: string;
  link: string;
  id: number;
  description: string;
  posted: Date;
  userEmail: string;
  files: FileVm[];
  type: string;
  creatorId: number;
  commentsPermissions: UserRoles;
  articleVisibilityPermissions: UserRoles;
  nickname: string;
  comments: CommentVm[];
}

File: Server/IBigDataPortal/EduLinks.Contracts/ViewModels/EduLinkViewModel.cs
using Files.Contracts.ViewModels;
using IBigDataPortal.Database.Entities;

namespace EduLinks.Contracts.ViewModels;

public class EduLinkViewModel
{
    public int Id { get; set; }
    public string Title { get; set; }
    public string Description { get; set; }
    public string UserEmail { get; set; }
    public string Nickname { get; set; }
    public string Type { get; set; }
    public DateTimeOffset Posted { get; set; } 
    public List<FileVm?> Files { get; set; } = new();
    public List<EduLinkCommentViewModel> Comments { get; set; } = new();

}